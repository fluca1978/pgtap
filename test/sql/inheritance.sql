\unset ECHO
--\i test/setup.sql

SELECT plan( 4 );

-- Create inherited tables
CREATE TABLE public.parent( id INT PRIMARY KEY );
CREATE TABLE public.child(  id INT PRIMARY KEY ) INHERITS ( public.parent );

-- Create inherited tables in another schema
CREATE SCHEMA hide;
CREATE TABLE hide.h_parent( id INT PRIMARY KEY );
CREATE TABLE hide.h_child(  id INT PRIMARY KEY ) INHERITS ( hide.h_parent );


-- test has_inhereted_tables
SELECT * FROM check_test(
       has_inherited_tables( 'hide'::name, 'h_parent'::name )
       , true  -- expected value
       , 'hide.h_parent is supposed to be parent of other tables'
       );

-- test hasnt_inherited_tables
SELECT * FROM check_test(
       hasnt_inherited_tables( 'hide'::name, 'h_child'::name )
       , true  -- expected value
       , 'hide.h_child is supposed not to have children'
);


-- test has_inhereted_tables
SELECT * FROM check_test(
       has_inherited_tables( 'parent'::name )
       , true  -- expected value
       , 'public.parent is supposed to be parent of other tables'
);

-- test hasnt_inherited_tables
SELECT * FROM check_test(
       hasnt_inherited_tables( 'child'::name )
       , true  -- expected value
       , 'public.child is supposed not to have children'
);





/****************************************************************************/
-- Finish the tests and clean up.
SELECT * FROM finish();
ROLLBACK;
