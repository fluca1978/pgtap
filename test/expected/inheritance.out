\unset ECHO
1..17
ok 1 - hide.h_parent is supposed to be parent of other tables should pass
ok 2 - hide.h_child2 is not supposed to have children should pass
ok 3 - public.parent is supposed to be parent of other tables should pass
ok 4 - public.child2 is supposed not to have children should pass
ok 5 - hide.h_parent direct is father of hide.h_child1 should pass
ok 6 - hide.h_parent direct is father of hide.h_child1 should pass
ok 7 - hide.h_child1 direct is father of hide.h_child2 should pass
ok 8 - hide.h_parent is father of hide.h_child2 should pass
ok 9 - child1 inherits from parent should pass
ok 10 - hide.h_parent is not father of public.child1 should fail
ok 11 - public.parent is not father of public.child1 should pass
ok 12 - hide.h_parent is not father of public.child1 should pass
ok 13 - parent is not father of public.child1 should fail
ok 14 - parent is not child1 should pass
ok 15 - child1 inherits from parent should fail
ok 16 - hide.h_parent is not child of hide.h_child1 should pass
ok 17 - hide.h_child1 inherits from hide.h_parent should fail